dll/win32/shell32/dialogs/drive.cpp

```diff
@@ -166,69 +166,69 @@ typedef struct _DRIVE_PROP_PAGE
     UINT DriveType;
 } DRIVE_PROP_PAGE;
 
-HRESULT
-SH_ShowDriveProperties(WCHAR *pwszDrive, LPCITEMIDLIST pidlFolder, PCUITEMID_CHILD_ARRAY apidl)
+BOOL
+SH_ShowDriveProperties(WCHAR *pwszDrive, IDataObject *pDataObj)
 {
     HPSXA hpsx = NULL;
     HPROPSHEETPAGE hpsp[MAX_PROPERTY_SHEET_PAGE];
-    PROPSHEETHEADERW psh;
     CComObject<CDrvDefExt> *pDrvDefExt = NULL;
-    WCHAR wszName[256];
 
-    ZeroMemory(&psh, sizeof(PROPSHEETHEADERW));
-    psh.dwSize = sizeof(PROPSHEETHEADERW);
-    psh.dwFlags = 0; // FIXME: make it modeless
-    psh.hwndParent = NULL;
-    psh.nStartPage = 0;
-    psh.phpage = hpsp;
+    CDataObjectHIDA cida(pDataObj);
+    if (FAILED_UNEXPECTEDLY(cida.hr()))
+        return cida.hr();
 
-    LPITEMIDLIST completePidl = ILCombine(pidlFolder, apidl[0]);
-    if (!completePidl)
-        return E_OUTOFMEMORY;
-
-    if (ILGetDisplayNameExW(NULL, completePidl, wszName, ILGDN_NORMAL))
+    RECT rcPosition = {CW_USEDEFAULT, CW_USEDEFAULT, 0, 0};
+    POINT pt;
+    if (SUCCEEDED(DataObject_GetOffset(pDataObj, &pt)))
     {
-        psh.pszCaption = wszName;
-        psh.dwFlags |= PSH_PROPTITLE;
+        rcPosition.left = pt.x;
+        rcPosition.top = pt.y;
     }
 
-    ILFree(completePidl);
+    DWORD style = WS_DISABLED | WS_CLIPSIBLINGS | WS_CAPTION;
+    DWORD exstyle = WS_EX_WINDOWEDGE | WS_EX_APPWINDOW;
+    CStubWindow32 stub;
+    if (!stub.Create(NULL, rcPosition, NULL, style, exstyle))
+    {
+        ERR("StubWindow32 creation failed\n");
+        return FALSE;
+    }
 
-    CComPtr<IDataObject> pDataObj;
-    HRESULT hr = SHCreateDataObject(pidlFolder, 1, apidl, NULL, IID_PPV_ARG(IDataObject, &pDataObj));
+    PROPSHEETHEADERW psh = {sizeof(PROPSHEETHEADERW)};
+    psh.dwFlags = PSH_PROPTITLE;
+    psh.pszCaption = pwszDrive;
+    psh.hwndParent = stub;
+    psh.nStartPage = 0;
+    psh.phpage = hpsp;
 
+    HRESULT hr = CComObject<CDrvDefExt>::CreateInstance(&pDrvDefExt);
     if (SUCCEEDED(hr))
     {
-        hr = CComObject<CDrvDefExt>::CreateInstance(&pDrvDefExt);
+        pDrvDefExt->AddRef(); // CreateInstance returns object with 0 ref count
```

Comment:
General question: Is this behaviour you mention in comment expected? (it's not a bug?)

Reply:
It is expected for `CComObject`, but unexpected in normal COM behavior, hence the comment.